sql/expression/function/greatest_least_test.go

```diff
@@ -0,0 +1,209 @@
+package function
+
+import (
+	"github.com/stretchr/testify/require"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+	"gopkg.in/src-d/go-mysql-server.v0/sql/expression"
+	"testing"
+)
+
+func TestGreatest(t *testing.T) {
+	t.Run("null", func(t *testing.T) {
+		require := require.New(t)
+		f, err := NewGreatest(
+			expression.NewLiteral(nil, sql.Null),
+			expression.NewLiteral(5, sql.Int64),
+			expression.NewLiteral(1, sql.Int64),
+		)
+		require.NoError(err)
+
+		v, err := f.Eval(sql.NewEmptyContext(), nil)
+		require.NoError(err)
+		require.Equal(nil, v)
+	})
+
+	t.Run("negative and all ints", func(t *testing.T) {
+		require := require.New(t)
+		f, err := NewGreatest(
+			expression.NewLiteral(int64(-1), sql.Int64),
+			expression.NewLiteral(int64(5), sql.Int64),
+			expression.NewLiteral(int64(1), sql.Int64),
+		)
+		require.NoError(err)
+
+		v, err := f.Eval(sql.NewEmptyContext(), nil)
+		require.NoError(err)
+		require.IsType(int64(1), v)
```

Comment:
this assert is redundant. require.Equal already fails if the type is not the same as the one of the expected value

Reply:
Fixed in next push.